src/plugins/event_annotation/public/components/group_editor_controls/group_editor_controls.tsx

```diff
@@ -0,0 +1,184 @@
+  TagSelector: SavedObjectsTaggingApiUiComponent['SavedObjectSaveModalTagSelector'];
+  dataViews: DataView[];
+  createDataView: (spec: DataViewSpec) => Promise<DataView>;
+  queryInputServices: QueryInputServices;
+}) => {
+  // save the spec for the life of the component since the user might change their mind after selecting another data view
+  const [adHocDataView, setAdHocDataView] = useState<DataView>();
+
+  useEffect(() => {
+    if (group.dataViewSpec) {
+      createDataView(group.dataViewSpec).then(setAdHocDataView);
+    }
+  }, [createDataView, group.dataViewSpec]);
+
+  const setSelectedAnnotation = useCallback(
+    (newSelection: EventAnnotationConfig) => {
+      update({
+        ...group,
+        annotations: group.annotations.map((annotation) =>
+          annotation.id === newSelection.id ? newSelection : annotation
+        ),
+      });
+      _setSelectedAnnotation(newSelection);
+    },
+    [_setSelectedAnnotation, group, update]
+  );
+
+  const dataViews = useMemo(() => {
+    const items = [...globalDataViews];
+    if (adHocDataView) {
+      items.push(adHocDataView);
+    }
+    return items;
+  }, [adHocDataView, globalDataViews]);
+
+  const currentDataView = useMemo(
+    () => dataViews.find((dataView) => dataView.id === group.indexPatternId) || dataViews[0],
+    [dataViews, group.indexPatternId]
+  );
+
+  return !selectedAnnotation ? (
+    <>
+      <EuiText
+        size="s"
+        css={css`
+          margin-bottom: ${euiThemeVars.euiSize};
+        `}
+      >
+        <h4>
+          <FormattedMessage id="eventAnnotation.groupEditor.details" defaultMessage="Details" />
+        </h4>
+      </EuiText>
+      <EuiForm>
+        <EuiFormRow
+          label={i18n.translate('eventAnnotation.groupEditor.title', {
+            defaultMessage: 'Title',
+          })}
+        >
+          <EuiFieldText
+            data-test-subj="annotationGroupTitle"
+            value={group.title}
+            onChange={({ target: { value } }) =>
+              update({
+                ...group,
+                title: value,
+              })
+            }
+          />
+        </EuiFormRow>
+        <EuiFormRow
+          label={i18n.translate('eventAnnotation.groupEditor.description', {
+            defaultMessage: 'Description',
+          })}
+        >
+          <EuiTextArea
+            data-test-subj="annotationGroupDescription"
+            value={group.description}
+            onChange={({ target: { value } }) =>
+              update({
+                ...group,
+                description: value,
+              })
+            }
+          />
+        </EuiFormRow>
+        <EuiFormRow>
+          <TagSelector
+            initialSelection={group.tags}
+            onTagsSelected={(tags: string[]) =>
+              update({
+                ...group,
+                tags,
+              })
+            }
+          />
+        </EuiFormRow>
```

Comment:
Can we mark both the description and tags as "Optional" (in a subdued text color) using the label append method we've used previously?

Reply:
Should this be added to EUI?